pkg/csi/cinder/controllerserver_test.go

```diff
@@ -260,6 +260,80 @@ func TestCreateVolumeWithIgnoreVolumeAZ(t *testing.T) {
 	assert.Equal("foo", actualRes.Volume.AccessibleTopology[0].GetSegments()[topologyKey])
 }
 
+// Test CreateVolume with --with-topology=false flag
+func TestCreateVolumeWithTopologyDisabled(t *testing.T) {
+	assert := assert.New(t)
+
+	tests := []struct {
+		name                string
+		volumeParams        map[string]string
+		expectedVolumeAZ    string
+		expectedCSITopology string
+	}{
+		{
+			name:             "empty paramters",
+			volumeParams:     nil,
+			expectedVolumeAZ: "",
+			// FIXME(stephenfin): This should be unset
+			expectedCSITopology: "nova",
+		},
+		{
+			name: "availability parameter",
+			volumeParams: map[string]string{
+				"availability": "cinder",
+			},
+			// FIXME(stephenfin): This should be "cinder" per the parameter
```

Comment:
Should this test, and the change for fixing the issues it highlights should be part of a different PR? Or perhaps provide the fix directly... I wonder what's easiest.

edit: ok, I see you use this new test in the next commit to validate it's not breaking behavior. So perhaps providing the fix into the same PR makes sense?

Reply:
I originally had the fix rolled in here, but Matt suggested it probably didn't belong in here as it was a behaviour change.

Reply:
Fair enough. Do I understand correctly that the plan is to create a new PR with the fix once this one merges?

Reply:
Yes, that's my plan